lib/hanami/model/sql/entity/schema.rb

```diff
@@ -38,6 +38,24 @@ def initialize(registry, relation, mapping)
             freeze
           end
 
+          # Process attributes
+          #
+          # @param attributes [#to_hash] the attributes hash
+          #
+          # @raise [TypeError] if the process fails
+          #
+          # @since 1.0.1
+          # @api private
+          def call(attributes)
+            Utils::Hash.new(
+              schema.call(attributes)
+            ).deep_symbolize!
```

Comment:
Sorry for the delay, RL got in the way. =/
 It's as @jodosha said, `deep_symbolize!` ends up being way too aggressive and it ends up calling `to_hash` on an entity when it's returned by the association.

It'd remove this method as it would probably ending creating more trouble while we do not "fix" `utils`.